includes/discovery/sensors/state/asa.inc.php

```diff
@@ -32,19 +40,23 @@
         create_state_index($state_name, $states);
 
         foreach ($temp as $index => $entry) {
-            $descr = ucwords(trim(preg_replace('/\s*\([^\s)]*\)/', '', $temp[$index]['cfwHardwareInformation'])));
+            $descr = ucwords(trim(preg_replace('/\s*\([^\s)]*\)/', '', $temp[$index]['CISCO-FIREWALL-MIB::cfwHardwareInformation'])));
+
             if ($index == 'netInterface') {
-                $index = 4;
+                $oid_index = 4;
             } elseif ($index == 'primaryUnit') {
-                $index = 6;
+                $oid_index = 6;
             } elseif ($index == 'secondaryUnit') {
-                $index = 7;
+                $oid_index = 7;
             }
+
+            $sensor_value = $temp[$index]['CISCO-FIREWALL-MIB::cfwHardwareStatusValue'];
+
             //Discover Sensors
-            discover_sensor(null, 'state', $device, $cur_oid . $index, $index, $state_name, $descr, 1, 1, null, null, null, null, $temp[$index][' cfwHardwareStatusValue'], 'snmp', $index);
```

Comment:
I wonder if the issue was purely the space in the variable being used here.

At least the code's cleaned up a bit so thanks @rudybroersma

Reply:
Yeah in retrospect I wondered the same. At least it's now migrated to SnmpQuery :)